lib/commands/misc/rmrole.js

```diff
@@ -0,0 +1,53 @@
+const { Command } = require('discord.js-commando');
+const { GENERAL_ID } = require('../../constants.js');
+const { getChannel } = require('./../../util.js');
+
+module.exports = class RemoveRoleCommand extends Command {
+	constructor(client) {
+		super(client, {
+			name: 'rmrole',
+			group: 'misc',
+			memberName: 'rmrole',
+			description: 'Remove a self-assigned role from yourself.',
+			details: '',
+			args: [
+				{
+					key: 'role',
+					prompt: 'Remove a role!',
+					type: 'string',
+					infinite: false,
+					wait: 1
+				}
+			]
+		});
+		this.queues = {};
+	};
+
+  async run(msg, args) {
+    const channel = this.client.channels.get(getChannel(REOL_CHAN_ID));
+    const role = msg.guild.roles.find(r => r.name == args.role);
+    const memberRole = msg.guild.roles.find(r => r.name == 'Member');
+
+    // Prevent removal of required roles (Can be updated in the future if more required roles are ever added).
+    if (args.role === "Member" || args.role === "member") {
+      msg.reply("You can't remove that role from yourself!");
+      return;
+    // Remove the role requested if the user is already a Member and has the provided role
+    }
+
+    // Checks that the roles provided by the user exists.
+    try {
+      let roleExists = msg.guild.roles.has(role.id);
+    } catch (err) {
+      msg.reply("That role doesn't exist! Remember that role names are case-sensitive. Check which roles you have by clicking/long-pressing on your profile picture by a message or in the member list.");
+      return;
+    }
+
+    if (msg.member.roles.has(memberRole.id) && msg.member.roles.has(role.id)) {
```

Comment:
I don't think we need to check for the `memberRole` here because non-members will not have any other roles anyway.